Datastore/src/Connection/Grpc.php

```diff
@@ -242,7 +242,7 @@ public function runQuery(array $args)
                 $query['filter'] = $this->convertFilterProps($query['filter']);
             }
 
-            if (isset($query['limit'])) {
+            if (isset($query['limit']) && is_int($query['limit'])) {
```

Comment:
@vishwarajanand why was this added? This seems like it could break existing implementations if they supplied something like `$query['limit'] = '1'`

Reply:
@bshaffer I mentioned this in point #1 in `PR description > Changes`:

> Under Grpc mode, when a Gql query string goes through `runQuery`, and a parsed query object is returned with correct `limit` array, it doesn't need to be modified in subsequent fetches. Added a check to skip that.

In Grpc mode, the **non-Gql** query provides`$query['limit']` as an `int` which needs to be parsed into an array, but your comment suggests we support int-like `string` also.

If so, we should change the `is_int` to `!is_array`

Reply:
@bshaffer  I raised another PR to fix this bug: https://github.com/googleapis/google-cloud-php/pull/5826